### ISSUE TITLE
Type Variable Position is miscomputed when data types are imported

### ISSUE THREAD

**Issue by nikivazou** [ucsd-progsys/liquidhaskell#1595]
```
module Map where 

data Map k v = Tip | Map k v (Map k v)

empty :: Map k v 
empty = Tip 
```

```
import Map 

data Foo a = Foo {fval :: Map a Int } 
{-@ data Foo a = Foo { fval :: Map a {v:Int | 0 <= v} } @-}

fEmp :: Foo a 
fEmp = Foo empty -- is UNSAFE
```

The error goes away when the [rtv_is_pol (ty_var_info α)](https://github.com/ucsd-progsys/liquidhaskell/blob/develop/src/Language/Haskell/Liquid/Constraint/Generate.hs#L817) is removed. 

So we should either 1/make the positivity check optional or 2/fix the way variable positions are computed in the imported data types.

**Comment by ranjitjhala** [comment]
I'd go for (2) fix the way the positions are computed in imported types.

Why is there a difference in how the positions are computed anyway?

**Comment by nikivazou** [comment]
On Thu, Jan 30, 2020, 19:00 Ranjit Jhala <notifications@github.com> wrote:

> I'd go for (2) fix the way the positions are computed in imported types.
>
> Why is there a difference in how the positions are computed anyway?
>

I wonder the same thing... Need to debug

**Comment by nikivazou** [comment]
This cannot just be fixed, will turn the positivity check optional. 

1. It is a good example that positive only variables should be able to get arbitrarily instantiated. e.g., in the empty map `empty :: Map k v` both `k` and `v` are positive. But, we would like to use the empty map as `Map {v:Int | 0 <= v} v`. (which cannot work with the positivity check on). 

2. More importantly, we turned this positivity check to make functional extensionality sound. But, (as supported by various POPL discussions) functional equality within refinement types  (`f = g`) is unsound is no many ways. So, this positivity check is not enough to fix it. So, I don't see a reason why to have this check and change the "default" liquid type inference semantics. 
In short, I believe we should remove the positivity check and disallow function equality, until we have a sound functional equality treatment.

**Comment by facundominguez** [comment]
The examples here's are passing in the latest LH. I'm closing this issue, but please reopen if it hasn't been fully resolved.
